Framework/API/src/Algorithm.cpp

```diff
@@ -1678,7 +1710,16 @@ void Algorithm::execNonMaster() {
         // This is the reverse cast of what is done in
         // cacheWorkspaceProperties(), so it should never fail.
         const Property &prop = dynamic_cast<Property &>(*wsProp);
-        setProperty(prop.name(), ws->cloneEmpty());
+        auto clone = ws->cloneEmpty();
+        // Currently we have not implemented a proper cloneEmpty() for all
+        // workspace types, in particular the abundance of Workspace2D subtypes,
+        // so we do a safety check here.
```

Comment:
The implementation is actually really trivial, I simply chose not to do it since as long as we have no algorithms with MPI support that use, say, `GroupingWorkspace` there is no reason to do so.